### ISSUE TITLE
Prefer tools.is_apple_os instead of settings.os check

### ISSUE THREAD

**Issue by prince-chrismc** [conan-io/hooks#255]
```suggestion
        if tools.is_apple_os(self.settings.os):
```

_Originally posted by @madebr in https://github.com/conan-io/conan-center-index/pull/3623#discussion_r531101411_

**Comment by Croydon** [comment]
I think the hooks have no way to post recommendations yet and I think that there are valid cases to only check of macOS but not iOS and the rest. Hence, I don't think we can write a hook for that right now.

In general, I think there can be many things where recommendations / hints / warnings would be great for hooks.

**Comment by SpaceIm** [comment]
Agree, while many `self.settings.os == "Macos"` should be replaced by `tools.is_apple_os(self.settings.os)`, there is no simple rule.

**Comment by prince-chrismc** [comment]
Recently I discovered, some hooks are warnings, perhaps this might be a suggestive warning?
